src/Client.php

```diff
@@ -160,16 +182,19 @@ protected function padCurlOptions(RequestInterface $request) : array
     {
         $options = $this->curlOptions;
 
-        $options[CURLOPT_RETURNTRANSFER] = true;
-        $options[CURLOPT_HEADER]         = true;
-        $options[CURLOPT_CUSTOMREQUEST]  = $request->getMethod();
-        $options[CURLOPT_URL]            = (string) $request->getUri();
-        $options[CURLOPT_POSTFIELDS]     = (string) $request->getBody();
+        $options[CURLOPT_RETURNTRANSFER] = $options[CURLOPT_RETURNTRANSFER] ?? true;
+        $options[CURLOPT_HEADER]         = $options[CURLOPT_HEADER] ?? true;
+        $options[CURLOPT_CUSTOMREQUEST]  = $options[CURLOPT_CUSTOMREQUEST] ?? $request->getMethod();
+        $options[CURLOPT_URL]            = $options[CURLOPT_URL] ?? (string) $request->getUri();
+        $options[CURLOPT_POSTFIELDS]     = $options[CURLOPT_POSTFIELDS] ?? (string) $request->getBody();
+        $options[CURLOPT_HTTPHEADER]     = $options[CURLOPT_HTTPHEADER] ?? [];
 
         foreach ($request->getHeaders() as $name => $values) {
             foreach ($values as $value) {
                 $header = sprintf('%s: %s', $name, $value);
-                $options[CURLOPT_HTTPHEADER][] = $header;
+                if (!in_array($header, $options[CURLOPT_HTTPHEADER])) {
+                    $options[CURLOPT_HTTPHEADER][] = $header;
+                }
             }
```

Comment:
here you are always going to update values, however there is no validation. Headers all have specifications via RFC. This may work for now but it could also hurt you.

Reply:
@Lewiscowles1986, thank you for your reply.

Actually the original approach doesn't have the validation verifies the headers.

And I think it should add the header validation to validate the header.

Could you suggest any header validation package?

Thanks.